src/domUpdates.js

```diff
@@ -136,9 +143,20 @@ class DomUpdates {
         ${instruction.instruction}</li>
         `)
     })
-    //add functionality for dropDown for ingredients to purchase
+    this.ingredientsPurchaseDropDown(allMissingIngredients)
   }
-    
+
+  ingredientsPurchaseDropDown(allMissingIngredients) {
```

Comment:
This was killer Bill - you knocked it out of the park on getting this set up.  Well done.